pulser-core/pulser/register/_reg_drawer.py

```diff
@@ -120,26 +136,44 @@ def _draw_2D(
                     else:
                         j += 1
                 # Sort qubits in plot_ids[i] according to masked status
+                det_map = [
+                    q for q, weight in dmm_qubits.items() if weight > 0.0
+                ]
                 plot_ids[i] = sorted(
                     plot_ids[i],
-                    key=lambda s: s in [str(q) for q in masked_qubits],
+                    key=lambda s: s in det_map,
                 )
-                # Merge all masked qubits
-                has_masked = False
+                # Merge all masked qubits with their detuning
+                # if the detunings are not all the same (masked qubits then)
+                has_det_map = False
+                is_mask = len(set([dmm_qubits[q] for q in det_map])) == 1
                 for j in range(len(plot_ids[i])):
-                    if plot_ids[i][j] in [str(q) for q in masked_qubits]:
-                        plot_ids[i][j:] = [", ".join(plot_ids[i][j:])]
-                        has_masked = True
+                    if plot_ids[i][j] in [str(q) for q in det_map]:
+                        qubit_det = [
+                            (
+                                q
+                                + (
+                                    f": {dmm_qubits[int(q)]:.2f}"
+                                    if not is_mask
+                                    else ""
+                                )
+                            )
+                            for q in plot_ids[i][j:]
+                        ]
```

Comment:
The formatting is making this very hard to read... Here's an alternative:
```suggestion
                        qubit_det = []
                        for q in plot_ids[i][j:]:
                            extra_label = f":{dmm_qubits[int(q)]:.2f}" if not is_mask else ""
                            qubit_det.append(f"q{extra_label}")
```
I can't tell if this will look better after formatting though